alchemy/bin/bootstrap-s3.ts

```
@@ -0,0 +1,262 @@
+            LocationConstraint: region,
+          },
+        }),
+    });
+
+    await s3Client.send(createCommand);
+    console.log(`✅ Created S3 bucket: ${bucketName}`);
+
+    // Add bootstrap tags
+    await s3Client.send(
+      new PutBucketTaggingCommand({
+        Bucket: bucketName,
+        Tagging: {
+          TagSet: [
+            {
+              Key: BOOTSTRAP_TAG_KEY,
+              Value: BOOTSTRAP_TAG_VALUE,
+            },
+            {
+              Key: "Purpose",
+              Value: "Alchemy state storage",
+            },
+            {
+              Key: "CreatedBy",
+              Value: "alchemy-cli",
+            },
+          ],
+        },
+      }),
+    );
+
+    console.log("✅ Tagged bucket with bootstrap markers");
+  } catch (error: any) {
+    if (error.name === "BucketAlreadyExists") {
+      throw new Error(
+        `Bucket name '${bucketName}' is already taken. Please try again to generate a new random suffix.`,
+      );
+    } else if (error.name === "BucketAlreadyOwnedByYou") {
+      console.log(`✅ Bucket ${bucketName} already exists and is owned by you`);
+
+      // Still add tags if they're missing
+      try {
+        await s3Client.send(
+          new PutBucketTaggingCommand({
+            Bucket: bucketName,
+            Tagging: {
+              TagSet: [
+                {
+                  Key: BOOTSTRAP_TAG_KEY,
+                  Value: BOOTSTRAP_TAG_VALUE,
+                },
+                {
+                  Key: "Purpose",
+                  Value: "Alchemy state storage",
+                },
+                {
+                  Key: "CreatedBy",
+                  Value: "alchemy-cli",
+                },
+              ],
+            },
+          }),
+        );
+        console.log("✅ Added bootstrap tags to existing bucket");
+      } catch (tagError: any) {
+        console.warn(
+          `⚠️  Could not add tags to existing bucket: ${tagError.message}`,
+        );
+      }
```

Do not swallow errors